MapboxNavigation/RouteMapViewController.swift

```diff
@@ -882,70 +874,6 @@ extension RouteMapViewController: StepsViewControllerDelegate {
-        NotificationCenter.default.addObserver(self, selector: #selector(RouteMapViewController.keyboardWillHide(notification:)), name: UIResponder.keyboardWillHideNotification, object: nil)
-    }
-
-    func unsubscribeFromKeyboardNotifications() {
-        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardWillShowNotification, object: nil)
-        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardWillHideNotification, object: nil)
-    }
-
-    @objc func keyboardWillShow(notification: NSNotification) {
-        guard self.navigationView.endOfRouteView != nil else { return }
-        guard let userInfo = notification.userInfo else { return }
-        guard let curveValue = userInfo[UIResponder.keyboardAnimationCurveUserInfoKey] as? Int else { return }
-        guard let duration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? Double else { return }
-        guard let keyBoardRect = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect else { return }
-
-        let curve = UIView.AnimationCurve(rawValue: curveValue) ?? UIView.AnimationCurve.easeIn
-        let options = (duration: duration, curve: curve)
-        let keyboardHeight = keyBoardRect.size.height
-
-        if #available(iOS 11.0, *) {
-            navigationView.endOfRouteShowConstraint?.constant = -1 * (keyboardHeight - view.safeAreaInsets.bottom) // subtract the safe area, which is part of the keyboard's frame
-        } else {
-            self.navigationView.endOfRouteShowConstraint?.constant = -1 * keyboardHeight
-        }
-
-        let opts = UIView.AnimationOptions(curve: options.curve)
-        UIView.animate(withDuration: options.duration, delay: 0, options: opts, animations: view.layoutIfNeeded, completion: nil)
-    }
-
-    @objc func keyboardWillHide(notification: NSNotification) {
-        guard self.navigationView.endOfRouteView != nil else { return }
-        guard let userInfo = notification.userInfo else { return }
-        let curve = UIView.AnimationCurve(rawValue: userInfo[UIResponder.keyboardAnimationCurveUserInfoKey] as! Int)
-        let options = (duration: userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as! Double,
-                       curve: UIView.AnimationOptions(curve: curve!))
-
-        self.navigationView.endOfRouteShowConstraint?.constant = 0
-
-        UIView.animate(withDuration: options.duration, delay: 0, options: options.curve, animations: view.layoutIfNeeded, completion: nil)
-    }
-}
-
-private extension UIView.AnimationOptions {
-    init(curve: UIView.AnimationCurve) {
-        switch curve {
-        case .easeIn:
-            self = .curveEaseIn
-        case .easeOut:
-            self = .curveEaseOut
-        case .easeInOut:
-            self = .curveEaseInOut
-        case .linear:
-            self = .curveLinear
-        @unknown default:
-            fatalError("Unknown curve")
```

Comment:
There was a crash here, fixed in 2bf7799f, but then since it was only used for the feedback views, I deleted it all anyway.